notation/notation.go

```diff
@@ -0,0 +1,314 @@
+package notation
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/signature"
+	"github.com/notaryproject/notation-core-go/timestamp"
+	"github.com/notaryproject/notation-go/registry"
+	"github.com/notaryproject/notation-go/verification/trustpolicy"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const annotationX509ChainThumbprint = "io.cncf.notary.x509chain.thumbprint#S256"
+
+// Descriptor describes the artifact that needs to be signed.
+type Descriptor struct {
+	// The media type of the targeted content.
+	MediaType string `json:"mediaType"`
+
+	// The digest of the targeted content.
+	Digest digest.Digest `json:"digest"`
+
+	// Specifies the size in bytes of the blob.
+	Size int64 `json:"size"`
+
+	// Contains optional user defined attributes.
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// Equal reports whether d and t points to the same content.
+func (d Descriptor) Equal(t Descriptor) bool {
+	return d.MediaType == t.MediaType && d.Digest == t.Digest && d.Size == t.Size && reflect.DeepEqual(d.Annotations, t.Annotations)
+}
+
+// SignOptions contains parameters for Signer.Sign.
+type SignOptions struct {
+	// Reference of the artifact that needs to be signed.
+	ArtifactReference string
+
+	// SignatureMediaType is the envelope type of the signature.
+	// Currently both `application/jose+json` and `application/cose` are
+	// supported.
+	SignatureMediaType string
+
+	// Expiry identifies the expiration time of the resulted signature.
+	Expiry time.Time
+
+	// TSA is the TimeStamp Authority to timestamp the resulted signature if
+	// present.
+	TSA timestamp.Timestamper
+
+	// TSAVerifyOptions is the verify option to verify the fetched timestamp
+	// signature.
+	// The `Intermediates` in the verify options will be ignored and
+	// re-contrusted using the certificates in the fetched timestamp signature.
+	// An empty list of `KeyUsages` in the verify options implies
+	// ExtKeyUsageTimeStamping.
+	TSAVerifyOptions x509.VerifyOptions
```

Comment:
Do we need to take this as input? Timestamps are as per RFC 3161 so we know what needs to be verified.

https://github.com/notaryproject/notaryproject/blob/main/specs/signature-specification.md#unsigned-attributes

Reply:
We can remove this option as it will be succeeded by the trust store and policy in the future.

Reply:
Removed.